sdk/tables/azure-tables/src/TableClient.ts

```diff
@@ -176,31 +177,38 @@ export class TableClient {
   }
 
   /**
-   * Update entity in the table.
-   * @param entity The properties of the updated entity.
-   * @param ifMatch Match condition for an entity to be updated. If specified and a matching entity is not found, an error will be raised. To force an unconditional update, set to the wildcard character (*). If not specified, an insert will be performed when no existing entity is found to update and a replace will be performed if an existing entity is found.
+   * Update an entity in the table.
+   * @param entity The properties of the entity to be updated.
+   * @param mode The different modes for updating the entity:
+   *             - Merge: Updates an entity by updating the entity's properties without replacing the existing entity.
+   *             - Replace: Updates an existing entity by replacing the entire entity.
+   * @param etag The ETag of the entity to be updated. If specified and a matching entity is not found, an error will be raised. To force an unconditional update, set to the wildcard character (*). If not specified, an insert will be performed when no existing entity is found to update and a replace will be performed if an existing entity is found.
    * @param options The options parameters.
    */
   updateEntity(
     entity: Entity,
-    ifMatch?: string,
+    mode: UpdateMode = "Merge",
```

Comment:
I guess we'd like to have both `etag` and `mode` inside UpdateEntityOptions. @ellismg, @xirzec, @bterlson?

Reply:
I think that `etag` going into options makes sense.  For mode, I think if we are going to provide a default of `Merge` as we do here, it should be in the options bag.  I could also imagine us not providing a default mode and keeping it as a parameter.

Reply:
That's a good point! I looked at the other languages and this is what they seem to be doing:
C# and Python - parameter default to "Merge"
Java - Required parameter.

So I think we either:

a) move it to options bag and default to "Merge"
b) Keep it as a required parameter without any defaults 

I think I like being explicit about the intent of the update (b). Do you guys have any strong preferences?